From f30f1adc1144a8864649fbda45c72e88184cd43f Mon Sep 17 00:00:00 2001
From: Franck Bui <fbui@suse.com>
Date: Tue, 30 Apr 2019 16:10:13 +0200
Subject: [PATCH 24/29] meson: make source files including nspawn-settings.h
 depend on libseccomp

Since nspawn-settings.h includes seccomp.h, any file that includes
nspawn-settings.h should depend on libseccomp so the correct header path where
seccomp.h lives is added to the header search paths.

It's especially important for distros such as openSUSE where seccomp.h is not
shipped in /usr/include but /usr/include/libseccomp.

This patch is similar to 8238423095ca54c48d9408a5da13e0325b31e6f6.

(cherry picked from commit 9f3f596477ba68d4ece645536586756caea4ae97)
---
 src/fuzz/meson.build   | 4 ++--
 src/nspawn/meson.build | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/fuzz/meson.build b/src/fuzz/meson.build
index 0d1ad2b1e1..a6c6db372b 100644
--- a/src/fuzz/meson.build
+++ b/src/fuzz/meson.build
@@ -128,12 +128,12 @@ fuzzers += [
         [['src/fuzz/fuzz-nspawn-settings.c'],
          [libshared,
           libnspawn_core],
-         []],
+         [libseccomp]],
 
         [['src/fuzz/fuzz-nspawn-oci.c'],
          [libshared,
           libnspawn_core],
-         []],
+         [libseccomp]],
 
         [['src/fuzz/fuzz-calendarspec.c'],
          [libshared],
diff --git a/src/nspawn/meson.build b/src/nspawn/meson.build
index 31217c7b46..c049ac6754 100644
--- a/src/nspawn/meson.build
+++ b/src/nspawn/meson.build
@@ -48,7 +48,7 @@ tests += [
         [['src/nspawn/test-nspawn-tables.c'],
          [libnspawn_core,
           libshared],
-         []],
+         [libseccomp]],
 
         [['src/nspawn/test-patch-uid.c'],
          [libnspawn_core,
-- 
2.21.0

